src/compiler/checker.ts

```diff
@@ -7413,7 +7409,7 @@ namespace ts {
             if (type.root.isDistributive) {
                 const simplified = getSimplifiedType(type.checkType);
                 const constraint = simplified === type.checkType ? getConstraintOfType(simplified) : simplified;
-                if (constraint) {
+                if (constraint && constraint !== type.checkType) {

```

Comment:
This particular change fixes an issue where a circularity resulted from the constraint of a conditional type being the conditional type itself. This can happen when a distributive conditional type is applied to an intersection of mapped types. The issue was uncovered because we now correctly defer resolution of conditional types involving generic mapped types.